### PR TITLE
feat(Box): add support for `transparent` background color

### PR DESCRIPTION
**Pull request by snitin315**
https://github.com/razorpay/blade/assets/46647141/51595886-6c29-46f5-8d6e-38e215468ff3

### CODE REVIEW

File: packages/blade/src/components/Box/Box.tsx [hunk 1]
```diff
@@ -12,7 +12,8 @@ const validateBackgroundString = (stringBackgroundColorValue: string): void => {
   if (__DEV__) {
     if (
       !stringBackgroundColorValue.startsWith('surface.background') &&
-      !stringBackgroundColorValue.startsWith('brand.')
+      !stringBackgroundColorValue.startsWith('brand.') &&
+      stringBackgroundColorValue !== 'transparent'
     ) {
       throwBladeError({
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Box/Box.tsx:18]: add transparent here as well in the message??

File: packages/blade/src/components/Box/__tests__/Box.web.test.tsx [hunk 2]
```diff
@@ -119,6 +119,32 @@ describe('<Box />', () => {
     `);
   });
 
+  it('should accept "transparent" value for backgroundColor', () => {
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Box/__tests__/Box.web.test.tsx:122]: I'm not sure if this test gives us something meaningful, ideally how do I check that transparent shall remove background and undefined shall apply the base?

Comment (snitin315) [reply packages/blade/src/components/Box/__tests__/Box.web.test.tsx:122]: This just verifies `background-color: 'transparent'` was applied, and can be removed.

Comment (snitin315) [reply packages/blade/src/components/Box/__tests__/Box.web.test.tsx:122]: >I check that transparent shall remove background and undefined shall apply the base?

We can see it from styles in the snap:

1. For `backgroundColor={{ base: 'brand.primary.300', l: undefined }}`:

```
.c0 {
        background-color: hsla(218,89%,51%,0.09);
      }
```

2. For `backgroundColor={{ base: 'brand.primary.300', l: 'transparent' }}`:

```
.c0 {
        background-color: hsla(218,89%,51%,0.09);
      }
      @media screen and (min-width:1024px) {
        .c0 {
          background-color: transparent;
        }
      }
```

File: packages/blade/src/components/Box/Box.tsx (hunk 1 above)
Comment (snitin315) [reply packages/blade/src/components/Box/Box.tsx:18]: Done.

**Review by saurabhdaware — APPROVED** [review]

**Review by anuraghazra — APPROVED** [review]

**Pull request merged**